design-documents/tracking-tool.md

```diff
@@ -0,0 +1,83 @@
+# Tech Upgrade Strategy
+
+## Overview
+
+Magento uses different 3rd party components and libraries like Symfony, Zend, hash generators, SDK, etc. And during Magento release we need to update components versions manually. Also, updates to the latest package version might require some code migration in Magento, which might impact release dates.
+To avoid such problem, we need to have a mechanism to track latest versions of 3rd party components.
+
+As most of dependencies specified in `composer.json`, such process can be automated.
+
+## Solution
+
+Dependencies, specified in `composer.json`, can be parsed and according to the dependencies list, we can find the latest version of the package.
```

Comment:
> Dependencies, specified in `composer.json`, can be parsed and according to the dependencies list, we can find the latest version of the package.

This seems to not account for the content under "Packages source" below:

> The most of packages are specified in `composer.json` but some of them are not specified, like database, or can be added in the future not via composer.
So we need to provide a possibility to extend source of packages versions.

> The tool should support multiple input types like CLI, configuration files, URL to composer.json file.

Reply:
Why? This section exactly explains why we need to support multiple sources and it's a child section of the "Solution" section